src/transformers/models/beit3/modeling_beit3.py

```diff
@@ -0,0 +1,1459 @@
+    def __init__(self, config):
+        super().__init__(config)
+        embed_dim = config.hidden_size
+        self.num_labels = config.num_labels
+        self.beit3 = Beit3Model(config, add_pooling_layer=True)
+
+        self.classifier = nn.Sequential(
+            nn.Linear(embed_dim, embed_dim * 2),
+            nn.LayerNorm(embed_dim * 2, eps=config.layer_norm_eps),
+            nn.GELU(),
+            nn.Linear(embed_dim * 2, config.num_labels),
+        )
+        self.post_init()
+
+    @add_start_docstrings_to_model_forward(BEIT3_FOR_VQA_INPUTS_DOCSTRING)
+    @replace_return_docstrings(output_type=SequenceClassifierOutput, config_class=_CONFIG_FOR_DOC)
+    def forward(
+        self,
+        input_ids,
+        pixel_values: torch.FloatTensor,
+        attention_mask: Optional[torch.Tensor] = None,
+        output_hidden_states: Optional[bool] = None,
+        output_attentions: Optional[bool] = None,
+        return_dict: Optional[bool] = None,
+        labels: Optional[torch.LongTensor] = None,
+    ) -> Union[Tuple[Any], SequenceClassifierOutput]:
+        r"""
+        Returns:
+
+        Examples:
+
+        ```python
+        >>> from transformers import Beit3Processor, Beit3ForQuestionAnswering
+        >>> from PIL import Image
+        >>> import requests
+
+        >>> processor = Beit3Processor.from_pretrained("Raghavan/beit3_base_patch16_480_vqa")
+        >>> model = Beit3ForQuestionAnswering.from_pretrained("Raghavan/beit3_base_patch16_480_vqa")
+
+        >>> url = "http://images.cocodataset.org/val2017/000000039769.jpg"
+        >>> image = Image.open(requests.get(url, stream=True).raw)
+        >>> question = "How many cats are there?"
+
+        >>> inputs = processor(text=question, images=image, return_tensors="pt")
+
+        >>> # forward pass
+        >>> outputs = model(**inputs)
+
+        >>> logits = outputs.logits
+        >>> # model predicts one of the 3129 possible answers
+        >>> predicted_answer_idx = logits.argmax(-1).item()
+        >>> predicted_answer = model.config.id2label[predicted_answer_idx]
+        >>> print("Predicted answer:", predicted_answer)
+        Predicted answer: 2
+        ```"""
+        return_dict = return_dict if return_dict is not None else self.config.use_return_dict
+
+        outputs = self.beit3(
+            input_ids=input_ids,
+            pixel_values=pixel_values,
+            attention_mask=attention_mask,
+            output_hidden_states=output_hidden_states,
+            output_attentions=output_attentions,
+            return_dict=return_dict,
+        )
+
+        pooled_output = outputs.pooler_output if return_dict else outputs[1]
+        logits = self.classifier(pooled_output)
+
+        loss = None
+        if labels is not None:
+            loss_fct = nn.KLDivLoss(reduction="batchmean")
+            log_softmax = nn.LogSoftmax(dim=-1)
+            logits = log_softmax(logits)
+            loss = loss_fct(logits, labels.contiguous())
```

Comment:
This part seems a bit weird. I think for VQAv2 standard loss is BceCrossEntropy and this is what Beit3 seems to use:
https://github.com/microsoft/unilm/blob/master/beit3/engine_for_finetuning.py#L201

Reply:
Feel free to adjust accordingly